src/plugins/github/translateContinuations.js

```diff
@@ -0,0 +1,334 @@
+  // GitHub schema says that it can. This warning is more of a
+  // diagnostic to the SourceCred maintainers (if it comes up on a real
+  // repository, we can learn something!) than an indication that
+  // something has gone wrong.
+  | {|+type: "NON_COMMIT_REF_TARGET", +target: GitObjectJSON|}
+  // This can happen if a commit has a parent that we did not fetch. We
+  // only fetch commits that are Git-reachable from HEAD or are the direct
+  // merge commit of a pull request. We may therefore omit commits that
+  // disappeared from master after a force-push, or were an ancestor of a
+  // pull request that was merged into a branch other than master. See
+  // issue #923 for more context. If this is omitted, we will simply
+  // omit the offending parent commit.
+  | {|+type: "UNKNOWN_PARENT_OID", +child: GitObjectID, +parent: GitObjectID|};
+
+export default function translate(
+  json: GithubResponseJSON
+): {|
+  +result: Repository,
+  +warnings: $ReadOnlyArray<Warning>,
+|} {
+  const repositoryJson = json.repository;
+  const warnings: Array<Warning> = [];
+
+  // Most of the work that this function does is exploding connections
+  // into lists of nodes. But commits require some special attention,
+  // because we have to resolve parent OIDs to actual parent commits.
+  // This means that it is most convenient to start by discovering all
+  // commits in the data.
+  const commits: Map<
+    GitObjectID,
+    {|
+      ...Commit,
+      parents: Array<null | Commit>, // mutable: we build this incrementally
+    |}
+  > = new Map();
+
+  // First, create all the commit objects, initializing them with empty
+  // parent arrays. We put these temporarily into a map keyed by OID for
+  // deduplication: a commit may appear both in the linearized history
+  // from HEAD and also as the merge commit of a pull request, and we
+  // want to process it just once.
+  const commitJsons: $ReadOnlyArray<CommitJSON> = Array.from(
+    new Map(
+      Array.from(
+        (function*() {
+          if (repositoryJson.defaultBranchRef) {
+            const target = repositoryJson.defaultBranchRef.target;
+            switch (target.__typename) {
+              case "Commit":
+                yield* target.history.nodes;
+                break;
+              case "Tree":
+              case "Blob":
+              case "Tag":
+                warnings.push({type: "NON_COMMIT_REF_TARGET", target});
+                break;
+              // istanbul ignore next: unreachable per Flow
+              default:
+                throw new Error((target.type: empty));
+            }
+          }
+          for (const pull of repositoryJson.pulls.nodes) {
+            if (pull.mergeCommit) {
+              yield pull.mergeCommit;
+            }
+          }
+        })()
+      ).map((json) => [json.oid, json])
+    ).values()
+  );
+  for (const commitJson of commitJsons) {
+    if (commits.has(commitJson.oid)) {
+      continue;
```

Comment:
I think this block is vestigial, b.c. you have already de-duplicated commitJsons by oid

Reply:
I think this comment is vestigial b.c. of GitHub shenanigans.